keras_cv/models/densenet.py

```diff
@@ -0,0 +1,296 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DenseNet models for KerasCV.
+
+Reference:
+  - [Densely Connected Convolutional Networks](https://arxiv.org/abs/1608.06993)
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+
+BN_AXIS = 3
+
+
+def dense_block(x, blocks, name):
+    """A dense block.
+
+    Args:
+      x: input tensor.
+      blocks: integer, the number of building blocks.
+      name: string, block label.
```

Comment:
If we intend for this block to be reusable, `name` should always default to None and should then be converted to a unique string based on the function name.

Reply:
How do you recommend we generate these names?  Will Keras automatically do this?

Reply:
I believe this is the way to do it: 

https://github.com/keras-team/keras/blob/b88f655b5fb81d157e1489cf4f3eff577a81203f/keras/applications/regnet.py#L403-L414

Reply:
done, thanks!